src/app/digital/views/components/XORGateView.ts

```diff
@@ -0,0 +1,78 @@
+import {DEFAULT_BORDER_COLOR, DEFAULT_BORDER_WIDTH, DEFAULT_CURVE_BORDER_WIDTH, DEFAULT_FILL_COLOR, GATE_NOT_CIRCLE_RADIUS, SELECTED_BORDER_COLOR, SELECTED_FILL_COLOR} from "core/utils/Constants";
```

Comment:
I think you can see that these classes are extraordinarily similar to each-other.
I think a good compromise would be to have just a `XORGateView` and it can take in an extra parameter in the constructor for `not`, which when `not` is true, it will draw the not circle